src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/StaticWebAssetsIntegrationTest.cs

```diff
@@ -54,6 +54,59 @@ public async Task Build_GeneratesStaticWebAssetsManifest_Success_CreatesManifest
             }
         }
 
+        [Fact]
+        [InitializeTestProject("AppWithPackageAndP2PReference")]
+        public async Task Publish_CopiesStaticWebAssetsToDestinationFolder()
+        {
+            var result = await DotnetMSBuild("Publish", "/restore");
+
+            Assert.BuildPassed(result);
+
+            // GenerateStaticWebAssetsManifest should generate the manifest and the cache.
+            Assert.FileExists(result, PublishOutputPath, Path.Combine("wwwroot", "_content", "PackageLibraryDirectDependency", "css", "site.css"));
+            Assert.FileExists(result, PublishOutputPath, Path.Combine("wwwroot", "_content", "PackageLibraryDirectDependency", "js", "pkg-direct-dep.js"));
+            Assert.FileExists(result, PublishOutputPath, Path.Combine("wwwroot", "_content", "PackageLibraryTransitiveDependency", "js", "pkg-transitive-dep.js"));
+        }
+
+        [Fact]
+        [InitializeTestProject("AppWithPackageAndP2PReference")]
+        public async Task Publish_Incremental_DoesNotCopyAnyFiles()
```

Comment:
This isn't super valuable. Publishing isn't something you do in a tight loop and users tend to clean the publish dir. I wouldn't test this. What would be useful to do a test similar to `PublishIntegrationTest.Publish_WithP2P_WorksWhenBuildProjectReferencesIsDisabled` which is how VS does publishing.

Reply:
Changed the test to simulate publish from visual studio